### PR TITLE
5 patch api reviews reviewid

### PR DESCRIPTION
**Pull request by ECJMac**
PATCH review by ID test completed and updated model & controller.
Error handling updated for GET & PATCH review by ID.

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -33,7 +33,31 @@ describe("app tests", () => {
...
+              category: "euro game",
+              created_at: "2021-01-18T10:00:20.514Z",
+              votes: 1,
+            });
+          });
+      });
+    });
+    describe("PATCH /api/reviews/:review_id", () => {
+      test("200: responds with updated review object where vote is incremented by newVote value", () => {
+        const REVIEW_ID = 1;
```
Comment (douglashellowell) [inline __tests__/app.test.js:56]: great use of `const` variables in tests

you should use this to match the `review_id` in the returned review too!

```diff
{
- review_id: 1,
+ review_id: REVIEW_ID,
}
```

🌟

**Review by douglashellowell — CHANGES_REQUESTED** [review]
great stuff!

Lots of changes in this PR related to `GET /api/reviews/:review_id` - try your best to keep these in their own PRs!


There are some more tests needed for `PATCH /api/reviews/:review_id`

think about 

```
/api/review/999999
/api/review/not-an-id

and

/api/review/1  { wrong-key: 1 }
```

**Comment by Dutton1990** [comment]
You could add one more test for api/review/99999 (for example), this should return a 404 not found as it's a valid id but doesn't exist :)

**Review by Angaca — APPROVED** [review]
Ready for merge!

**Pull request merged**